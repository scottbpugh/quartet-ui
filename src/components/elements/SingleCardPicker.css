/* Copyright (c) 2018 SerialLab Corp.

GNU GENERAL PUBLIC LICENSE
   Version 3, 29 June 2007

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>. */



/* Auto-sized cards */
.pick-cards-container {
  display: flex;
  flex-wrap:wrap;
  margin:0;
  padding:0;
  justify-content:center;
  align-items:center;
  position:relative;
}


.pick-cards-container .card-picker {
  display:flex;
  flex-direction: column;
  margin:20px 20px;
  background: #f2f2f2;
  color: #444;
  border-radius: 2px;
  padding:60px 20px 10px;
  position:relative;
  height:330px;
  width:300px;
  text-align:center;
  overflow:hidden;
  border: 1px solid #CCC;
  transition: transform .2s; /* Animation */
}

.pick-cards-container .small-card-picker {
  height:auto;
  padding-bottom:0;
}

.pick-cards-container .card-picker:hover {
    cursor: pointer;
    transform: scale(1.08); /* (150% zoom - Note: if the zoom is too large, it will go outside of the viewport) */
}
.pt-dark .pick-cards-container .card-picker {
    box-shadow: 6px 6px 6px rgba(0, 0, 0, 0.4);
    border: 0px;
}

.pick-cards-container .card-picker h5 {
  background: #0E5A8A;
  top:0;
  position: absolute;
  width: 100%;
  left:0;
  color:#FFF;
  border-radius: 0;
  margin-bottom:20px;
  border-top-left-radius:2px;
  border-top-right-radius:2px;
  padding:15px 10px;
  text-align:center;
}

.picker-data-list {
  list-style:none;
  margin:0;
  padding:0;
}

.picker-data-list li {
  margin:5px;
  padding:5px;
  border-bottom:1px solid #DDD;
}
.picker-data-list li:last-child {
  margin:5px;
  padding:5px;
  border-bottom: none;
}
