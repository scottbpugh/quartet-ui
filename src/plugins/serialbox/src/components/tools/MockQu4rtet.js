// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
// Fake local storage.
const qu4rtetRequire = require("qu4rtet-ui/lib/require.js").default;

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value;
  }

  removeItem(key) {
    delete this.store[key];
  }
}
window = global;
export const localStorage = new LocalStorageMock();

// add fake local storage for tests.
window.localStorage = localStorage;

// add fake require for ipcRenderer.
window.require = module => {
  if (module === "path") {
    return {
      join: () => {
        return "";
      }
    };
  } else if (module === "electron") {
    return {
      ipcRenderer: {
        on: () => {},
        send: () => {}
      },
      remote: {
        require: path => {
          if (path === "./main-process/plugin-manager.js") {
            return {
              install: () => {
                return {done: "ok"};
              }
            };
          }
          return {};
        },
        app: {
          getVersion: () => {
            return "test-version";
          },
          getPath: () => {
            return "somepath";
          }
        }
      }
    };
  } else {
    try {
      return qu4rtetRequire(module);      
    } catch(e) {
      return null;
    }

  }
};

const {qu4rtet} = require("qu4rtet-ui/lib/qu4rtet");
const React = qu4rtetRequire("react");
const {Component} = React;
