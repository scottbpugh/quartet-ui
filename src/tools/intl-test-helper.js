// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React from "react";
import renderer from "react-test-renderer";
import {IntlProvider} from "react-intl";

import messages from "../messages";
import {flattenMessages} from "../lib/flattenMessages";

const returnComponentWithIntl = (
  children,
  props = {locale: "en-US", messages: flattenMessages(messages["en-US"])}
) => {
  return (
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );
};

export default returnComponentWithIntl;
