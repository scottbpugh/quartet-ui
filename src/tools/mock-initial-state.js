export default {
  dashboard: {
    notifications: [
      {
        id: "93d2a5c2-0936-4798-80e2-14c3e8df2d96",
        type: "error",
        msg: "An EPCIS document with invalid GTIN EPC URN could not be parsed."
      },
      {
        id: "e3660e15-6884-4668-8401-5da19399472d",
        type: "warning",
        msg:
          "An outbound EPCIS job to 172.112.10.27 timed out. A retry will be performed in 14mn30s."
      },
      {
        id: "4314dade-aefd-49cb-b695-1add8a6a1785",
        type: "warning",
        msg: "User lduros was locked out after too many invalid token attempts."
      }
    ]
  },
  serversettings: {
    servers: {
      "d0246781-67c6-474b-8ab0-29de61b6e6bb": {
        serverID: "d0246781-67c6-474b-8ab0-29de61b6e6bb",
        protocol: "http",
        port: "8000",
        path: "",
        ssl: false,
        hostname: "localhost",
        serverSettingName: "box 1",
        url: "http://localhost:8000/",
        appList: ["", "capture", "epcis", "manifest", "rest-auth", "serialbox"],
        username: "lduros",
        password: "password1234"
      }
    }
  },
  form: {
    addRegion: {
      values: {
        active: false
      }
    }
  },
  intl: {
    defaultLocale: "en-US",
    locale: "en-US",
    messages: {
      "app.nav.servers": "Servers",
      "app.nav.numberRange": "Number Ranges",
      "app.nav.dashboard": "Dashboard",
      "app.nav.server": "Server",
      "app.nav.plugins": "Plugins",
      "app.serverSettings.serverSettings": "Server Settings",
      "app.serverSettings.addAServer": "Add a New Server",
      "app.serverSettings.serverSettingsSaved":
        "Your server settings were saved",
      "app.serverSettings.serverDeleted": "Server removed successfully",
      "app.themes.lightTheme": "Light Theme",
      "app.themes.darkTheme": "Dark Theme",
      "app.themes.contrastedTheme": "Contrasted Theme",
      "app.themes.darkBrownTheme": "Dark Brown Theme",
      "app.themes.polarTheme": "Polar Theme",
      "app.servers.addServer": "Add Server",
      "app.servers.updateServer": "Update Server",
      "app.servers.registerUser": "Register User",
      "app.servers.verifyUser": "Verify User",
      "app.servers.userCreated": "User {username} successfully created.",
      "app.servers.userVerified": "User successfully verified.",
      "app.servers.deleteServer": "Remove Server",
      "app.servers.deleteServerConfirm":
        "Are you sure you want to remove this server? No data from this server will be deleted.",
      "app.servers.noServerMsg":
        "You currently have no QU4RTET server in your configuration. Click the + icon above and add a new server.",
      "app.servers.errorFormFetch":
        "An error occurred while attempting to get this form from remote server {serverName}. {error}",
      "app.servers.errorServerFetch":
        "An error occurred while requesting initial data from server {serverName}. Please check your settings and credentials. {error}",
      "app.plugins.addPlugin": "Add a Plugin",
      "app.plugins.pluginEnabled": "Plugin enabled",
      "app.plugins.pluginDisabled": "Plugin disabled",
      "plugins.numberRange.pool": "Pool",
      "plugins.numberRange.region": "Region",
      "plugins.numberRange.allocation": "Allocation",
      "plugins.numberRange.allocateButton": "Allocate from Pool",
      "plugins.numberRange.createdOn": "Created On",
      "plugins.numberRange.readableName": "Readable Name",
      "plugins.numberRange.machineName": "Machine Name",
      "plugins.numberRange.status": "Status",
      "plugins.numberRange.requestThreshold": "Request Threshold",
      "plugins.numberRange.regions": "Regions",
      "plugins.numberRange.numberRangePools": "Number Range Pools",
      "plugins.numberRange.active": "active",
      "plugins.numberRange.inactive": "inactive",
      "plugins.numberRange.range": "Range",
      "plugins.numberRange.to": "to",
      "plugins.numberRange.state": "State",
      "plugins.numberRange.addSequentialRegion": "Add a New Sequential Region",
      "plugins.numberRange.editSequentialRegion": "Edit Sequential Region",
      "plugins.numberRange.addRandomizedRegion": "Add a New Randomized Region",
      "plugins.numberRange.editRandomizedRegion": "Edit Randomized Region",
      "plugins.numberRange.addPool": "Add a New Pool",
      "plugins.numberRange.editPool": "Edit Pool",
      "plugins.numberRange.noRegionInPool":
        "There is currently no region in this pool.",
      "plugins.numberRange.regionDetailTitle": "Pool {poolName} Regions",
      "plugins.numberRange.navItemsTitle": "Serial Number Pools",
      "plugins.numberRange.current": "current",
      "plugins.numberRange.serial": "Sequential",
      "plugins.numberRange.randomized": "Randomized",
      "plugins.numberRange.deleteRegion": "Delete region {regionName}",
      "plugins.numberRange.allocatedSuccess":
        "{size} allocated to region {regionName}. You will be prompted to save the export file shortly.",
      "plugins.numberRange.regionDeletedSuccessfully":
        "Region deleted successfully",
      "plugins.numberRange.deleteRegionConfirm":
        "Are you sure you want to delete this region?",
      "plugins.numberRange.errorFetchPools":
        "An error occurred while attempting to fetch pools from {serverName}",
      "plugins.numberRange.errorVanilla":
        "An error occurred while performing this operation. {error}",
      "plugins.numberRange.errorFetchPool":
        "An error occurred while attempting to fetch {poolName}. {error}",
      "plugins.numberRange.errorFetchRegion":
        "An error occurred while attempting to get region information. {error}",
      "plugins.numberRange.errorAllocating":
        "An error occurred while attempting to allocate from pool {poolName}. {error}",
      "plugins.numberRange.errorFormFetch":
        "An error occurred while attempting to get this form from remote server {serverName}. {error}",
      "plugins.numberRange.errorFailedToGenerateFile":
        "An error occurred when attempting to generate an export file from allocation."
    }
  },
  layout: {
    pageTitle: {
      id: "nav.app.serverDetails",
      defaultMessage: "Server Details",
      values: {}
    },
    currentPath: "/server-details/d0246781-67c6-474b-8ab0-29de61b6e6bb",
    theme: "dark-brown",
    location: {
      pathname: "/server-details/d0246781-67c6-474b-8ab0-29de61b6e6bb",
      search: "",
      hash: "",
      key: "x0qo0w"
    }
  },
  plugins: {
    navTreeItems: {
      plugin_NumberRange_NumberRangeNavRoot: {
        pluginName: "NumberRange",
        pluginComponentName: "plugin_NumberRange_NumberRangeNavRoot"
      }
    },
    plugins: {
      NumberRange: {
        enabled: true
      }
    }
  },
  numberrange: {
    servers: {
      "704e4478-f018-4fb0-b0b3-2711bbdd325c": {
        pools: [
          {
            sequentialregion_set: [
              "http://localhost:8000/serialbox/sequential-region-detail/blob/"
            ],
            randomizedregion_set: [],
            created_date: "2018-03-14T15:44:06.717881Z",
            modified_date: "2018-03-14T15:44:06.717912Z",
            readable_name: "test",
            machine_name: "test",
            active: true,
            request_threshold: 50000
          }
        ],
        server: {
          serverID: "704e4478-f018-4fb0-b0b3-2711bbdd325c",
          protocol: "http",
          port: "8000",
          path: "",
          ssl: false,
          hostname: "localhost",
          serverSettingName: "box 1",
          url: "http://localhost:8000/",
          appList: [
            "",
            "capture",
            "epcis",
            "manifest",
            "rest-auth",
            "serialbox"
          ],
          username: "admin",
          password: "password1234"
        }
      },
      "d6734acf-b19f-465d-aa55-741497051d17": {
        pools: [
          {
            sequentialregion_set: [
              "http://localhost:8000/serialbox/sequential-region-detail/blob/"
            ],
            randomizedregion_set: [],
            created_date: "2018-03-14T15:44:06.717881Z",
            modified_date: "2018-03-14T15:44:06.717912Z",
            readable_name: "test",
            machine_name: "test",
            active: true,
            request_threshold: 50000
          }
        ],
        server: {
          serverID: "d6734acf-b19f-465d-aa55-741497051d17",
          protocol: "http",
          port: "8000",
          path: "",
          ssl: false,
          hostname: "localhost",
          serverSettingName: "Box 1",
          url: "http://localhost:8000/",
          appList: [
            "",
            "capture",
            "epcis",
            "manifest",
            "rest-auth",
            "serialbox"
          ],
          username: "admin",
          password: "password1234"
        }
      },
      "d8f53d6a-eaaa-4129-b3b5-d14723086f48": {
        pools: [
          {
            sequentialregion_set: [],
            randomizedregion_set: [],
            created_date: "2018-03-14T15:44:06.717881Z",
            modified_date: "2018-03-14T15:44:06.717912Z",
            readable_name: "test",
            machine_name: "test",
            active: true,
            request_threshold: 50000
          }
        ],
        server: {
          serverID: "d8f53d6a-eaaa-4129-b3b5-d14723086f48",
          protocol: "http",
          port: "8000",
          path: "",
          ssl: false,
          hostname: "localhost",
          serverSettingName: "box 1",
          url: "http://localhost:8000/",
          appList: [
            "",
            "capture",
            "epcis",
            "manifest",
            "rest-auth",
            "serialbox"
          ],
          username: "admin",
          password: "password1234"
        }
      },
      "d0246781-67c6-474b-8ab0-29de61b6e6bb": {
        pools: [
          {
            sequentialregion_set: [],
            randomizedregion_set: [
              "http://localhost:8000/serialbox/randomized-regions/random2/",
              "http://localhost:8000/serialbox/randomized-regions/random24/"
            ],
            created_date: "2018-03-19T18:23:08.964545Z",
            modified_date: "2018-03-19T18:23:08.964578Z",
            readable_name: "docker pool",
            machine_name: "dockerpool",
            active: true,
            request_threshold: 50000
          }
        ],
        server: {
          serverID: "d0246781-67c6-474b-8ab0-29de61b6e6bb",
          protocol: "http",
          port: "8000",
          path: "",
          ssl: false,
          hostname: "localhost",
          serverSettingName: "box 1",
          url: "http://localhost:8000/",
          appList: [
            "",
            "capture",
            "epcis",
            "manifest",
            "rest-auth",
            "serialbox"
          ],
          username: "lduros",
          password: "password1234"
        }
      }
    },
    currentRegions: [
      {
        pool: "dockerpool",
        created_date: "2018-03-19T18:24:46.200566Z",
        modified_date: "2018-03-20T16:50:23.186915Z",
        readable_name: "random1",
        machine_name: "random2",
        active: true,
        order: 2,
        min: 1,
        max: 10,
        start: 1,
        current: 6,
        remaining: -10
      },
      {
        pool: "dockerpool",
        created_date: "2018-03-19T18:25:10.786631Z",
        modified_date: "2018-03-19T18:25:10.786661Z",
        readable_name: "random 2",
        machine_name: "random24",
        active: true,
        order: 3,
        min: 11,
        max: 200,
        start: 173,
        current: 173,
        remaining: 189
      }
    ],
    allocationDetail: {
      numbers: [7, 9, 8, 4],
      fulfilled: true,
      type: "random",
      encoding: "decimal",
      region: "random2",
      size_granted: 4
    }
  }
};
