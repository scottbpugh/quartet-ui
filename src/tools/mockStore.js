// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React from "react";

import {IntlProvider} from "react-intl";
import {addLocaleData} from "react-intl";
import messages from "messages";
import en from "react-intl/locale-data/en";
import fr from "react-intl/locale-data/fr";
import {flattenMessages} from "lib/flattenMessages";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

addLocaleData([...en, ...fr]);
let defaultLocale = "en-US";
const middlewares = [thunk];

export const initialState = {
  dashboard: {notifications: []},
  serversettings: {
    servers: {
      "824ae2dd-2411-4adf-a846-d9b310a33d4b": {
        serverSettingName: "Server #1",
        serverName: "localhost",
        port: "80",
        ssl: false,
        username: "root",
        password: "toor",
        serverID: "824ae2dd-2411-4adf-a846-d9b310a33d4b"
      }
    }
  },
  intl: {
    defaultLocale: defaultLocale,
    locale: defaultLocale,
    messages: flattenMessages(messages[defaultLocale])
  },
  layout: {pageTitle: {id: "app.nav.servers"}},
  plugins: {plugins: {}, navTreeItems: []}
};

export const mockStore = configureStore(middlewares);

export const TestWrapper = ({locale, messages, children}) => {
  if (locale === undefined) {
    locale = defaultLocale;
  }
  if (messages === undefined) {
    messages = initialState.intl.messages;
  }
  return (
    <IntlProvider locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
};

