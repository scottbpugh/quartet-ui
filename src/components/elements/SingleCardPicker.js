// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import {Button, ControlGroup, InputGroup} from "@blueprintjs/core";
import {FormattedMessage} from "react-intl";
import {withRouter} from "react-router";
import "./SingleCardPicker.css";
import {pluginRegistry} from "plugins/pluginRegistration";

/*
  Displays a list of objects (entries, events, companies, locations) as
  single cards.
*/
class _SingleCardPicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filter: "",
      keywordSearch: "",
      entries: [],
      entriesPerPage: 20,
      inputSize: 50,
      maxPages: 1
    };
    this.offset = 0;
    this.currentPage = 1;
    this.debounced = null;
  }

  // filter by a field in the rows.
  filterBy = evt => {
    this.setState({filter: evt.currentTarget.value}, () => {
      this.offset = 0;
      this.currentPage = 1;
      this.processEntries();
    });
  };

  // search by a field in the rows or all of them.
  searchBy = evt => {
    this.setState({keywordSearch: evt.currentTarget.value}, () => {
      this.offset = 0;
      this.currentPage = 1;
      this.processEntries();
    });
  };

  componentDidMount() {
    this.processEntries();
    this.setState({
      entries: this.props.entries,
      maxPages: 1
    });
  }

  // refresh the Lists, keeping the search filters.
  componentWillReceiveProps(nextProps) {
    let maxPages = this.currentPage;
    if (nextProps.next !== null && Array.isArray(nextProps.entries)) {
      maxPages = Math.ceil(nextProps.count / nextProps.entries.length);
    }
    this.setState({
      entries: nextProps.entries,
      maxPages: maxPages
    });
  }

  goTo = path => {
    this.props.history.push(path);
  };

  // go to next page if possible.
  next = () => {
    this.currentPage += 1;
    this.processEntries(true);
  };

  // go to previous page if possible.
  previous = () => {
    this.currentPage -= 1;
    this.processEntries(true);
  };

  processEntries = (clear = false) => {
    if (this.debounced) {
      clearTimeout(this.debounced);
    }
    this.debounced = setTimeout(() => {
      const {loadEntries, server} = this.props;
      loadEntries(
        server,
        this.state.keywordSearch,
        this.currentPage,
        this.props.ordering ? this.props.ordering : null,
        this.props.type ? this.props.type : null
      );
    }, clear ? 0 : 250);
  };

  render() {
    const {entries} = this.state;
    return (
      <div>
        <div className="table-control">
          <div className="pagination-control">
            <div>
              <Button
                disabled={this.currentPage - 1 < 1}
                onClick={this.previous.bind(this)}>
                previous
              </Button>{" "}
              |{" "}
              <Button
                disabled={this.currentPage >= this.state.maxPages}
                onClick={this.next.bind(this)}>
                next
              </Button>
            </div>
          </div>
          <div>
            <ControlGroup fill={false} vertical={false}>
              <div className="pt-select">
                <select value={this.state.filter}>
                  <option value="">Search</option>
                </select>
              </div>
              <InputGroup
                onChange={this.searchBy}
                value={this.state.keywordSearch}
                placeholder={pluginRegistry
                  .getIntl()
                  .formatMessage({id: "app.common.enterKeywords"})}
              />
            </ControlGroup>
            <div className="label-info-display">
              <FormattedMessage
                id="app.common.entriesTotal"
                values={{entriesCount: this.props.count}}
              />
            </div>
          </div>
        </div>
        <div className="pick-cards-container">
          {Array.isArray(entries) && entries.length > 0
            ? entries.map((entry, index) => {
                return (
                  <div
                    className="card-picker"
                    key={`entry-card-${index}`}
                    style={this.props.entryStyle ? this.props.entryStyle : {}}>
                    <this.props.entryClass
                      {...this.props}
                      entry={entry}
                      server={this.props.server}
                      history={this.props.history}
                    />
                  </div>
                );
              })
            : null}
        </div>
      </div>
    );
  }
}

export const SingleCardPicker = withRouter(_SingleCardPicker);
window.qu4rtet.exports("components/elements/SingleCardPicker", this);
