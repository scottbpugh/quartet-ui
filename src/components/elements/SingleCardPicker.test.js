// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React from "react";
import "tools/mockStore"; // mock ipcRenderer, localStorage, ...
import renderer from "react-test-renderer";
import {TestWrapper} from "tools/mockStore";
import Enzyme, {mount, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import sinon from "sinon";
import toJson from "enzyme-to-json";
import {SingleCardPicker} from "./SingleCardPicker";

it("renders correctly", () => {
  const TestEntry = props => {
    const updateFieldVal = entry => {
      // trigger a redux form field value change
      props.changeValue(entry);
    };
    return (
      <div
        key={props.entry.id}
        onClick={updateFieldVal.bind(this, props.entry)}
      >
        <h5 className="bp3-heading">
          {props.entry.name}
        </h5>
        <ul className="picker-data-list">
          <li>
            {props.entry.GLN13}
          </li>
          {props.entry.city ? (
            <li>
              {props.entry.city}
            </li>
          ) : null}
          {props.entry.country ? (
            <li>
              {props.entry.country}
            </li>
          ) : null}
        </ul>
      </div>
    );
  };
  const CardPicker = renderer
    .create(
      <TestWrapper>
        <SingleCardPicker
          listTitle="Hello Test"
          changeValue={() => {}}
          loadEntries={() => {}}
          server={{serverID: "aServerID"}}
          entries={[
            {id: "anyID", GLN13: "anyvalue", city: "Falaise", country: "France"}
          ]}
          count={0}
          next={0}
          entryClass={TestEntry}
        />
      </TestWrapper>
    )
    .toJSON();
  expect(CardPicker).toMatchSnapshot();
});
