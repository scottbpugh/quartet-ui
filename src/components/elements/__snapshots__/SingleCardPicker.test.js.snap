// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div
      className="table-control"
    >
      <div
        className="pagination-control"
      >
        <div>
          <button
            className="pt-button pt-disabled"
            disabled={true}
            onClick={undefined}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span>
              previous
            </span>
          </button>
           
          |
           
          <button
            className="pt-button pt-disabled"
            disabled={true}
            onClick={undefined}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span>
              next
            </span>
          </button>
        </div>
      </div>
      <div>
        <div
          className="pt-control-group"
        >
          <div
            className="pt-select"
          >
            <select
              value=""
            >
              <option
                value=""
              >
                Search
              </option>
            </select>
          </div>
          <div
            className="pt-input-group"
          >
            <input
              className="pt-input"
              onChange={[Function]}
              placeholder="Enter Keywords..."
              style={
                Object {
                  "paddingRight": 0,
                }
              }
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="label-info-display"
        >
          <span>
            Entries Total
          </span>
        </div>
      </div>
    </div>
    <div
      className="pick-cards-container"
    >
      <div
        className="card-picker"
        style={Object {}}
      >
        <div
          onClick={[Function]}
        >
          <h5 />
          <ul
            className="picker-data-list"
          >
            <li>
              anyvalue
            </li>
            <li>
              Falaise
            </li>
            <li>
              France
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
